backstory/src/test/java/com/google/sps/images/AnnotatedImageTest.java

```diff
@@ -0,0 +1,177 @@
+    ByteArrayOutputStream bos = new ByteArrayOutputStream();
+
+    try {
+      BufferedImage bImage = ImageIO.read(new File(reference));
+      ImageIO.write(bImage, fileType, bos);
+    } catch (IOException exception){
+      throw exception;
+    }
+
+    return bos.toByteArray();
+  }
+
+  /**
+   * public AnnotatedImage(byte[] rawImageData, List<EntityAnnotation> labelAnnotations);
+   *  rawImageData is null.
+   *  labelAnnotations is empty.
+   *  Expects IllegalArgumentException to be thrown.
+  */
+  @Test(expected = IllegalArgumentException.class)
+  public void constructorNullImageData() throws IllegalArgumentException {
+    AnnotatedImage annotatedImageExpectingException = new AnnotatedImage(nullRawImageData, emptyLabelAnnotations);
+  }
+
+  /**
+   * public AnnotatedImage(byte[] rawImageData, List<EntityAnnotation> labelAnnotations);
+   *  rawImageData is empty.
+   *  labelAnnotations is non-empty.
+   *  Expects IllegalArgumentException to be thrown.
+  */
+  @Test(expected = IllegalArgumentException.class)
+  public void constructorEmptyImageData() throws IllegalArgumentException {
+    List<EntityAnnotation> labelAnnotations = new ArrayList<>();
+    labelAnnotations.add(makeMockEntityAnnotationWithDescription("Description")); 
+
+    AnnotatedImage annotatedImageExpectingException = new AnnotatedImage(emptyRawImageData, labelAnnotations);
+  }
+
+  /**
+   * public AnnotatedImage(byte[] rawImageData, List<EntityAnnotation> labelAnnotations);
+   *  rawImageData is non-empty image data.
+   *  labelAnnotations is non-empty.
+  */
+  @Test
+  public void constructorRawImageData() throws IllegalArgumentException, IOException {
+    List<EntityAnnotation> labelAnnotations = new ArrayList<>();
+    labelAnnotations.add(makeMockEntityAnnotationWithDescription("Description"));
+    byte[] rawImageData;
+    try {
+      rawImageData = getBytesFromImageReference(
+        "src/test/java/com/google/sps/images/data/dogRunningOnBeach.jpg", "jpg");
+    } catch (IOException exception) {
+      throw exception;
+    }
+
+    AnnotatedImage actualAnnotatedImage = new AnnotatedImage(rawImageData, labelAnnotations);
+    assertTrue(Arrays.equals(rawImageData, actualAnnotatedImage.getRawImageData()));
+    assertEquals(labelAnnotations, actualAnnotatedImage.getLabelAnnotations());
+  }
+
+  /**
+   * public List<String> getLabelDescriptions();
+   *  labelAnnotations is empty.
+  */
+  @Test
+  public void getLabelDescriptionsEmptyLabelAnnotations() throws IllegalArgumentException, IOException {
+    byte[] rawImageData;
+    try {
+      rawImageData = getBytesFromImageReference(
+        "src/test/java/com/google/sps/images/data/dogRunningOnBeach.jpg", "jpg");
+    } catch (IOException exception) {
+      throw exception;
+    }
+        
+    AnnotatedImage annotatedImageActual = new AnnotatedImage(rawImageData, emptyLabelAnnotations);
```

Comment:
Since there's only one image here, consider naming: `annotatedImage`

The additional "actual" adjective could be useful to differentiate this instance from another AnnotatedImage, but since there is no other, it doesn't contribute to clarity.

Reply:
updated!